lib/flakes.nix

```diff
@@ -19,4 +19,24 @@ rec {
     in
     self;
 
+  /* Generates an attrset for the given list of system strings and instantiates
+    nixpkgs from a given flake. The system architecture string for each set is
+    available via the 'system' argument. The instantiated nixpkgs set is available
+    via the 'pkgs' argument.
+
+    Type: forSystems :: [String] -> Flake -> ( String -> AttrSet -> t ) -> AttrSet
+
+    Example:
+      nixpkgs.lib.forSystems
+        [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ]
+        (builtins.getFlake "github:nixos/nixpkgs")
+        (system: pkgs: { x = pkgs.hello; }}
+
+      The example will return:
+        { aarch64-darwin = { ... }; aarcH64-linux = { ... }; x86_64-darwin = { ... }; x86_64-linux = { ... }; }
+  */
+  forSystems = systems: flake: f:
```

Comment:
The name does not really say what it does, maybe something like `legacyPackagesForSystems`.

Reply:
That is true, I will rename it.

Reply:
Fixed!